src/app/treasure-hunt/operation-costs/operation-costs.component.ts

```diff
@@ -278,5 +295,77 @@ export class OperationCostsComponent implements OnInit {
     }
   }
 
+  setOtherFuelCo2SavingsData() {
+    if (!this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData) {
+      let co2SavingsData: Co2SavingsData = {
+        energyType: 'fuel',
+        energySource: 'Petroleum-based fuels',
+        fuelType: 'Motor Gasoline',
+        totalEmissionOutputRate: 70.22,
+        electricityUse: 0,
+        eGridRegion: '',
+        eGridSubregion: '',
+        totalEmissionOutput: 0,
+        userEnteredBaselineEmissions: false,
+        userEnteredModificationEmissions: false,
+        zipcode: ''
+      }
+      this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData = co2SavingsData;     
+    }
+    if(!this.treasureHunt.currentEnergyUsage.otherFuelMixedCO2SavingsData){
+        this.treasureHunt.currentEnergyUsage.otherFuelMixedCO2SavingsData = new Array<Co2SavingsData>();
+      }
+    this.getOtherFuelOptions();
+  }
+
+  getOtherFuelOptions(){
+    if (this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.fuelType) {
+      let tmpOtherFuel: OtherFuel = _.find(this.otherFuels, (val) => { return this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.energySource === val.energySource; });
+      this.fuelOptions = tmpOtherFuel.fuelTypes;
+    }
+  }
+
+  setFuelOptions() {
+    let tmpOtherFuel: OtherFuel = _.find(this.otherFuels, (val) => { return this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.energySource === val.energySource; });
+    this.fuelOptions = tmpOtherFuel.fuelTypes;
+    this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.fuelType = undefined;
+    this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.totalEmissionOutputRate = undefined;
+    this.save();
+  }
+
+  setFuel() {
+    let tmpFuel: { fuelType: string, outputRate: number } = _.find(this.fuelOptions, (val) => { return this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.fuelType === val.fuelType; });
+    this.treasureHunt.currentEnergyUsage.otherFuelCO2SavingsData.totalEmissionOutputRate = tmpFuel.outputRate;
+    this.save();
+  }
+
+  
+  showMixedCO2EmissionsModal() {
+    this.treasureHuntService.modalOpen.next(true);
+    this.mixedCO2EmissionsModal.show();
+  }
+
+  hideMixedCO2EmissionsModal() {
+    this.treasureHuntService.modalOpen.next(false);
+    this.mixedCO2EmissionsModal.hide();
+  }
+
+  applyMixedCO2EmissionsModalData(mixedOutputRate: number) {
```

Comment:
These two 'apply__' methods do different things but are named similarly. Use better descriptive naming - make sure the event emitters from the modal make sense with whatever changes you make